### PR TITLE
MESOS: in scheduler create nodes with ready condition

### PR DESCRIPTION
**Pull request by sttts**
The current kube-ui version (compare https://github.com/kubernetes/kube-ui/pull/37/files#r46950087)
assumes that the first node condition is the Ready condition. This PR works
around this by enforcing the Ready condition to be the first one.

A parallel PR against kube-ui is in the works: https://github.com/kubernetes/kube-ui/pull/44

Fixes https://github.com/mesosphere/kubernetes-mesos/issues/669

### CODE REVIEW

**Comment by s-urbaniak** [comment]
LGTM

**Comment by sttts** [comment]
@k8s-bot test this

**Comment by sttts** [comment]
@k8s-bot unit test this

**Comment by sttts** [comment]
Rebased in the hope the tests are not flaky with newer master.

File: contrib/mesos/pkg/node/node.go [hunk 1]
```diff
@@ -55,6 +55,17 @@ func Create(
 		},
 		Status: api.NodeStatus{
 			Phase: api.NodePending,
+			// WORKAROUND(sttts): make sure that the Ready condition is the
```
Comment (jdef) [inline contrib/mesos/pkg/node/node.go]: workaround .. is there a more desirable solution? if so, should we file an issue to track it?

Comment (sttts) [inline contrib/mesos/pkg/node/node.go]: We can wait until a new kube-ui is out and the fix merged. But we will move this into the controller manager soon anyway, then this workaround goes away but itself. Moreover, having this behavior (without the kube-ui comment) would be completely reasonable even without this very reason.

**Comment by sttts** [comment]
@k8s-bot test this

**Comment by jdef** [comment]
OK - if this code is going away when we move it into the controller
manager, then let's say that in the comment. If the code has value on its
own (e.g. completely reasonable) then let's get rid of "WORKAROUND".

**Comment by sttts** [comment]
@jdef done.

I am holding this PR back though because I am optimistic that we have a kube-ui v4 soon:  https://github.com/kubernetes/kube-ui/pull/44. In the worst case, we can cherry-pick this PR on the v0.7.1 branch if it takes longer than expected.

**Comment by sttts** [comment]
As kube-ui v4 seems to be stalled, let's merge this for the time being.

**Pull request merged**